PLANET-7701: Fix main nav menu accessibility issues

mardelnet opened the pull request:
### Summary
This PR enhances the experience of the main navigation menu by:
- Displaying the Navigation Menu drop-downs with keyboard navigation ([see reference](https://www.apple.com/)).
- Adding a focus ring to the reset button of the search field.

---

Ref: https://jira.greenpeace.org/browse/PLANET-7701

### Testing
1. Navigate the main nav menu by tabbing its elements. It should mimic the behavior of [this menu](https://www.apple.com/). 
2. Check the reset button of the search fields has a visible focus state.

planet-4 commented:
### Test instance is ready :rocket:

:new_moon: [tavros](https://www-dev.greenpeace.org/test-tavros) | [admin](https://www-dev.greenpeace.org/test-tavros/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-tavros/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-tavros) | [composer-local.json](https://github.com/greenpeace/planet4-test-tavros/blob/main/composer-local.json)

:watch: 2025.02.14 09:44:23

GP-Dan-Tovbein commented:
I've teted the assigned instance and it seems to be working fine however I've added a subitem to the Donate nav and the current implementation it's not working as expected.

See the attached video
https://github.com/user-attachments/assets/3638891e-0455-412a-8151-15db9b15cb95

mardelnet commented on assets/src/js/accessible_nav_menu.js:
```diff
@@ -0,0 +1,115 @@
+/**
+ * Class to handle keyboard accessibility in the navigation menu.
+ */
+class AccessibleNavMenu {
```
You're right. Besides I love to use constants, in this case, it was a bit too much. Fixed!

GP-Dan-Tovbein commented on assets/src/scss/layout/_navbar.scss:
```diff
@@ -262,12 +262,41 @@ a.nav-link:hover:before,
+
+  .accessible-nav-link {
+    @include dropdown-icon;
+    right: 15px;
```
Same here!

mardelnet replied:
Done!

mardelnet commented on assets/src/scss/layout/_navbar.scss:
```diff
@@ -262,12 +262,41 @@ a.nav-link:hover:before,
+  padding: 0;
+  top: 0;
+  bottom: 0;
+  right: -15px;
```
Done!

mardelnet commented:
> I've teted the assigned instance and it seems to be working fine however I've added a subitem to the Donate nav and the current implementation it's not working as expected.
> 
> See the attached video https://github.com/user-attachments/assets/3638891e-0455-412a-8151-15db9b15cb95

Good catch, @GP-Dan-Tovbein 
I've already fixed it: https://www-dev.greenpeace.org/test-oberon/

GP-Dan-Tovbein commented on images/cancel.svg:
Cannot we use [this icon](https://github.com/greenpeace/planet4-master-theme/blob/93ca96578cfa1b12f660a8addd1432f7aec7a3d2/images/black-cross.svg?plain=1) instead?

mardelnet replied:
Good idea! Done!

GP-Dan-Tovbein commented on assets/src/scss/layout/_navbar.scss:
```diff
@@ -294,6 +323,27 @@ a.nav-link:hover:before,
+  width: 10px;
+  height: 10px;
+  pointer-events: auto;
+  outline: 2px solid #99c3ff;
```
Do you think if we can add this color as a variable or a token instead? Maybe talk with Houssam 🤔

mardelnet replied:
Good idea! I will open a new ticket/PR for this as it's not part of this task scope.

GP-Dan-Tovbein replied:
I think he can makes it in the meantime since it's not mandatory to touch your code. It can be done in parallel and then you can just use the token

GP-Dan-Tovbein commented on assets/src/scss/layout/navbar/_search.scss:
```diff
@@ -146,12 +147,23 @@
   width: 24px;
+  height: 24px;
+  background-color: transparent;
+  background-image: url("../../images/cancel.svg");
```
Do you think if we can replicate [this implementation](https://github.com/greenpeace/planet4-master-theme/blob/93ca96578cfa1b12f660a8addd1432f7aec7a3d2/assets/src/scss/layout/navbar/_burger-menu.scss#L221)? I mean, usin `mask`

mardelnet replied:
I don't think so. Maybe I'm wrong, but there seems to be an incompatibility between using `mask` and `outline` since the first one "hides" everything around it, making the outline invisible.

GP-Dan-Tovbein commented on templates/desktop-menu.twig:
```diff
@@ -19,8 +19,8 @@
                     </a>
-
                     {% if item.children is not empty %}
+                        <button class="accessible-nav-link" tabindex="0"></button>
```
Please check https://www.w3.org/WAI/standards-guidelines/act/rules/97a4e1/ if this implementation follow the test cases. I suspect we need to add a content. What do you think?

mardelnet replied:
Nice catch! Fixed!

GP-Dan-Tovbein reviewed (commented):
I've left some comments! Looking forward to your opinion!

GP-Dan-Tovbein reviewed (approved):
Code looks good to me! Excelent work 👏  

Please follow up the border color by using tokens 🙏
